File: 1282-group-the-people-given-the-group-size-they-belong-to/1282-group-the-people-given-the-group-size-they-belong-to.java
import java.util.*;
class Solution {
    public List<List<Integer>> groupThePeople(int[] arr) {
        List<List<Integer>> ans = new ArrayList<>();
        HashMap<Integer,List<Integer>> map = new HashMap<>();
        for (int i=0; i<arr.length; i++){
            int num = arr[i];
            if (!map.containsKey(num)){
                List<Integer> ls = new ArrayList<>();
                ls.add(i);
                map.put(num,ls);
            }
            else {
                
                if (map.get(num).size() == num){
                    ans.add(map.get(num));
                    map.remove(num);
                    List<Integer> ls = new ArrayList<>();
                    ls.add(i);
                    map.put(num,ls);
                }
                else{
                    map.get(num).add(i);
                }
            }
        }
        for (int l : map.keySet()){
            ans.add(map.get(l));
        }
        return ans;
    }
}

File: 1282-group-the-people-given-the-group-size-they-belong-to/1282-group-the-people-given-the-group-size-they-belong-to.cpp
class Solution {
public:
    vector<vector<int>> groupThePeople(vector<int>& nums) {
        set<int> s;
        
        for(int i=0; i<nums.size(); i++)
        {
            s.insert(nums[i]);
        }
        auto it = s.begin();
        vector<int> a;
        vector<vector<int>> ans;
        for(int i=0; i<s.size(); i++)
        {
            int count = 0;
            for(int j=0; j<nums.size();j++)
            {
                if(*it == nums[j])
                {
                    a.push_back(j);
                    count++;
                    if(count == *it)
                    {
                        ans.push_back(a);
                        a.clear();
                        count = 0;
                    }
                }
            }
            it++;
        }
        return ans;
    }
};